prov/lnx/src/lnx_ep.c

```diff
@@ -565,604 +429,170 @@ struct fi_ops_cm lnx_cm_ops = {
-	struct lnx_rx_entry *entry = (struct lnx_rx_entry *) item;
-
-	/* entry refers to the receive request waiting for a message
-	 * entry->rx_entry.tag is the tag passed in by the application.
-	 *
-	 * entry->rx_entry.addr is the address passed in by the application.
-	 * This is the LNX level address. It's only significant if it's set
-	 * to ADDR_UNSPEC. Otherwise, it has already been used to look up the
-	 * peer.
-	 *
-	 * entry->rx_cep is always NULL in this case, as this will only be
-	 * known when the message is received.
-	 *
-	 * entry->rx_peer is the LNX peer looked up if a valid address is
-	 * given by the application, otherwise it's NULL.
-	 *
-	 * match_attr information is filled by the lnx_get_tag() callback and
-	 * contains information passed to us by the core endpoint receiving
-	 * the message.
-	 *
-	 * match_attr->rx_peer is not significant because at the lnx_get_tag()
-	 * call there isn't enough information to find what the peer is.
-	 *
-	 * match_attr->lm_tag, match_attr->lm_ignore are the tag and ignore
-	 * bits passed up by the core endpoint receiving the message.
-	 *
-	 * match_attr->lm_addr is the address of the peer which sent the
-	 * message. Set if the core endpoint has done a reverse lookup,
-	 * otherwise set to ADDR_UNSPEC.
-	 *
-	 * match_attr->lm_cep is the core endpoint which received the message.
-	 */
-	return lnx_match_common(entry->rx_entry.tag, match_attr->lm_tag,
-			entry->rx_ignore, match_attr->lm_addr,
-			entry->rx_entry.addr, entry->rx_peer, match_attr->lm_cep);
-}
-
-static inline int
-lnx_init_queue(struct lnx_queue *q, dlist_func_t *match_func)
-{
-	int rc;
-
-	rc = ofi_spin_init(&q->lq_qlock);
-	if (rc)
-		return rc;
-
-	dlist_init(&q->lq_queue);
-
-	q->lq_match_func = match_func;
-
-	return 0;
-}
-
-static inline int
-lnx_init_qpair(struct lnx_qpair *qpair, dlist_func_t *recvq_match_func,
-			   dlist_func_t *unexq_match_func)
-{
-	int rc = 0;
-
-	rc = lnx_init_queue(&qpair->lqp_recvq, recvq_match_func);
-	if (rc)
-		goto out;
-	rc = lnx_init_queue(&qpair->lqp_unexq, unexq_match_func);
-	if (rc)
-		goto out;
-
-out:
-	return rc;
-}
-
-static inline int
-lnx_init_srq(struct lnx_peer_srq *srq)
+static int
+lnx_open_core_eps(struct lnx_ep *lep, void *context, size_t fclass)
 {
-	int rc;
-
-	rc = lnx_init_qpair(&srq->lps_trecv, lnx_match_recvq, lnx_match_unexq);
-	if (rc)
-		return rc;
-	rc = lnx_init_qpair(&srq->lps_recv, lnx_match_recvq, lnx_match_unexq);
-	if (rc)
-		return rc;
-
-	return rc;
-}
+	int i, rc = 0;
+	struct lnx_core_domain *cd;
+	struct lnx_core_ep *cep;
 
-static int lnx_get_ctx(struct local_prov_ep *ep, size_t fclass,
-		       struct fid_ep ***ep_ctx, size_t *size)
-{
-	switch (fclass) {
-	case FI_CLASS_RX_CTX:
-		*ep_ctx = ep->lpe_rxc;
-		*size = ep->lpe_fi_info->ep_attr->rx_ctx_cnt;
-		break;
-	case FI_CLASS_TX_CTX:
-		*ep_ctx = ep->lpe_txc;
-		*size = ep->lpe_fi_info->ep_attr->tx_ctx_cnt;
-		break;
-	default:
+	if (fclass != FI_CLASS_EP)
```

Comment:
now that I think of it, since you removed the sep path, you shouldn't even need this check, it should always be FI_CLASS_EP. I think you can remove the param entirely.

Reply:
done